### PR TITLE
fix: adding iam engine option to SM and prefix to SCC

### PR DESCRIPTION
**Pull request by iamar7**
### Description

Resolves: https://github.com/terraform-ibm-modules/stack-retrieval-augmented-generation/issues/12, https://github.com/terraform-ibm-modules/stack-retrieval-augmented-generation/issues/130

### Release required?

- [ ] No release
- [x] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

File: ibm_catalog.json [hunk 1]
```diff
@@ -264,6 +264,13 @@
 							"description": "The CRN of an existing secret manager instance to use in this solution. If not set, a new secret manager instance is provisioned.",
 							"required": false
 						},
+						{
+							"key": "iam_engine_enabled",
```
[the remark this answers is not in the document]
Comment (iamar7) [reply ibm_catalog.json:268]: I actually added this but somehow it was deleted in the recent [commit](https://github.com/terraform-ibm-modules/stack-retrieval-augmented-generation/pull/137/commits/61f20ef04e0b9cbc0d622190f59213e98c764353), so will revert the deletion.

**Comment by ocofaigh** [comment]
I'm going to close this issue because the RAG stack is being updated to consume the genai stack, and the genai stack consumes the core security services stack. Updating of RAG stack to consume genai stack is being tracked here https://github.ibm.com/goldeneye/issues/issues/9896 (internally)

**Pull request closed without merging**